servicetalk-transport-netty-internal/src/test/java/io/servicetalk/transport/netty/internal/ChannelSetTest.java

```diff
@@ -74,18 +73,22 @@
 
     @BeforeEach
     public void setupMocks() {
-        when(channel.id()).thenReturn(channelId);
-        when(channel.closeFuture()).thenReturn(channelCloseFuture);
+        lenient().when(channel.id()).thenReturn(channelId);
+        lenient().when(channel.closeFuture()).thenReturn(channelCloseFuture);
         lenient().when(channel.close()).then(invocation -> {
             listener.operationComplete(channelCloseFuture);
             return channelCloseFuture;
         });
-        when(channelCloseFuture.channel()).thenReturn(channel);
+        lenient().when(channelCloseFuture.channel()).thenReturn(channel);
         lenient().when(channel.pipeline()).thenReturn(channelPipeline);
         lenient().when(channel.attr(eq(CHANNEL_CLOSEABLE_KEY))).thenReturn(mockClosableAttribute);
         lenient().when(mockClosableAttribute.getAndSet(any())).thenReturn(nettyConnection);
+        lenient().when(nettyConnection.onClose()).thenReturn(fromSource(closeAsyncCompletable));
         lenient().when(nettyConnection.closeAsync()).thenReturn(fromSource(closeAsyncCompletable));
+        lenient().when(nettyConnection.closeAsyncNoOffload()).thenReturn(fromSource(closeAsyncCompletable));
```

Comment:
You don't need to mock offloaded methods anymore and you now need to validate that `closeAsyncGracefullyNoOffload()` is not invoked when not expected:
```patch
Index: servicetalk-transport-netty-internal/src/test/java/io/servicetalk/transport/netty/internal/ChannelSetTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/servicetalk-transport-netty-internal/src/test/java/io/servicetalk/transport/netty/internal/ChannelSetTest.java b/servicetalk-transport-netty-internal/src/test/java/io/servicetalk/transport/netty/internal/ChannelSetTest.java
--- a/servicetalk-transport-netty-internal/src/test/java/io/servicetalk/transport/netty/internal/ChannelSetTest.java	(revision a4b7d6b549df2b3a9a410fad69aaed41aa209220)
+++ b/servicetalk-transport-netty-internal/src/test/java/io/servicetalk/transport/netty/internal/ChannelSetTest.java	(date 1623269996694)
@@ -26,6 +26,7 @@
 import io.netty.channel.DefaultChannelId;
 import io.netty.util.Attribute;
 import io.netty.util.concurrent.GenericFutureListener;
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
@@ -83,10 +84,7 @@
         lenient().when(channel.pipeline()).thenReturn(channelPipeline);
         lenient().when(channel.attr(eq(CHANNEL_CLOSEABLE_KEY))).thenReturn(mockClosableAttribute);
         lenient().when(mockClosableAttribute.getAndSet(any())).thenReturn(nettyConnection);
-        lenient().when(nettyConnection.onClose()).thenReturn(fromSource(closeAsyncCompletable));
-        lenient().when(nettyConnection.closeAsync()).thenReturn(fromSource(closeAsyncCompletable));
         lenient().when(nettyConnection.closeAsyncNoOffload()).thenReturn(fromSource(closeAsyncCompletable));
-        lenient().when(nettyConnection.closeAsyncGracefully()).thenReturn(fromSource(closeAsyncGracefullyCompletable));
         lenient().when(nettyConnection.closeAsyncGracefullyNoOffload())
                 .thenReturn(fromSource(closeAsyncGracefullyCompletable));
         when(channelCloseFuture.addListener(any())).then((invocation) -> {
@@ -96,6 +94,13 @@
         fixture.addIfAbsent(channel);
     }
 
+    @AfterEach
+    public void finalValidation() {
+        // ChannelSet uses only "NoOffload" variant of these methods:
+        verify(nettyConnection, never()).onClose();
+        verify(nettyConnection, never()).closeAsync();
+        verify(nettyConnection, never()).closeAsyncGracefully();
+    }
+
     @Test
     void closeAsync() {
         Completable completable = fixture.closeAsync();
@@ -110,6 +115,7 @@
     void closeAsyncGracefullyWithNettyConnectionChannelHandler() throws Exception {
         Completable completable = closeAsyncGracefully(fixture, 100, SECONDS);
         verify(nettyConnection, never()).closeAsyncGracefully();
+        verify(nettyConnection, never()).closeAsyncGracefullyNoOffload();
         TestCompletableSubscriber subscriber = new TestCompletableSubscriber();
         toSource(completable).subscribe(subscriber);
         verify(nettyConnection).closeAsyncGracefullyNoOffload();
@@ -165,8 +171,6 @@
 
         TestCompletableSubscriber subscriber2 = new TestCompletableSubscriber();
         toSource(gracefulCompletable).subscribe(subscriber2);
-        verify(nettyConnection, never()).closeAsyncGracefully();
-
         fixture.onClose().toFuture().get();
 
         subscriber.awaitOnComplete();

```